images/inference-services/kfserving/run.sh

```diff
@@ -37,25 +40,63 @@ case $PREDICTOR in
     # named '1', create it and copy the tensorflow model to it.
     tensorflow)
         if [ "$(mc find ${model_bucket} --name 1)" = "" ]; then
-            mc cp -r ${model_bucket}/tfmodel/ ${model_bucket}/1
+            mc cp -r "${model_bucket}"/tfmodel/ "${model_bucket}"/1
         fi
-        export RUNTIME_VERSION=$(mc cat ${model_bucket}/requirements.txt | awk -F '=' '/tensorflow/ {print $3}')
+        export RUNTIME_VERSION=$(mc cat "${model_bucket}"/requirements.txt | awk -F '=' '/tensorflow/ {print $3}')
         prediction_url_path="${isvc}:predict"
         ;;
     # kfserving expects the sklearn model file as model.joblib however mlflow
     # saves the model as model.pkl, so if there is no model.joblib, create a
     # copy named model.joblib from model.pkl
     sklearn)
-        if ! mc ls ${model_bucket} | grep -q "model.joblib"; then
-            mc cp ${model_bucket}/model.pkl ${model_bucket}/model.joblib
+        if ! mc ls "${model_bucket}" | grep -q "model.joblib"; then
+            mc cp "${model_bucket}"/model.pkl "${model_bucket}"/model.joblib
         fi
         export PROTOCOL_VERSION="v2"
         prediction_url_path="${isvc}/infer"
         ;;
+    triton)
+        # triton supports multiple serving backends, each has its own directory
+        # structure (see: https://github.com/triton-inference-server/server/blob/main/docs/model_repository.md).
+        export PROTOCOL_VERSION="v2"
+        export RUNTIME_VERSION="21.08-py3"
+        export ARGS="[--strict-model-config=false]"
+        export FUSEML_MODEL="${FUSEML_MODEL}/triton"
+        prediction_url_path="${isvc}/infer"
+        if [ "$(mc find ${model_bucket} --name triton)" = "" ]; then
+            mlmodel="$(mc cat ${model_bucket}/MLmodel)"
+            if echo "${mlmodel}" | grep -q saved_model; then
+                mc cp -r "${model_bucket}"/tfmodel/ "${model_bucket}"/triton/"${isvc}"/1/model.savedmodel
+            elif echo "${mlmodel}" | grep -q onnx; then
+                model_file="$(echo ${mlmodel} | awk '/data:/ { print $2; exit }')"
+                mc cp -r "${model_bucket}"/"${model_file}" "${model_bucket}"/triton/"${isvc}"/1/
+            else
+                echo "ERROR: Only tensorflow 'saved model' nad ONNX format is supported"
```

Comment:
typo: `nad` should be `and`. And you should probably use _are_, plural here.

Reply:
fixed, thanks!